server/src/test/java/org/elasticsearch/index/mapper/DateFieldTypeTests.java

```diff
@@ -214,4 +220,36 @@ public void testRangeQuery() throws IOException {
                 () -> ft.rangeQuery(date1, date2, true, true, null, null, null, context));
         assertEquals("Cannot search on field [field] since it is not indexed.", e.getMessage());
     }
+
+    public void testDateNanoDocValues() throws IOException {
+        // Create an index with some docValues
+        Directory dir = newDirectory();
+        IndexWriter w = new IndexWriter(dir, new IndexWriterConfig(null));
+        Document doc = new Document();
+        NumericDocValuesField docValuesField = new NumericDocValuesField("my_date", 1444608000000L);
+        doc.add(docValuesField);
+        w.addDocument(doc);
+        docValuesField.setLongValue(1459641600000L);
+        w.addDocument(doc);
+        w.flush();
```

Comment:
no need to flush, `DirectoryReader.open` takes care of writing NRT segments for data that only lives in the IW buffers